storage/src/test/java/tech/pegasys/teku/storage/server/kvstore/BlindedBlockMigrationTest.java

```diff
@@ -51,33 +55,83 @@ public class BlindedBlockMigrationTest {
   private final DatabaseContext fullBlockContext =
       new DatabaseContext(DatabaseVersion.DEFAULT_VERSION, BlockStorage.FULL_BLOCK, false);
 
+  private final StubAsyncRunner stubAsyncRunner = new StubAsyncRunner();
+
   @Test
   void shouldMigrateHotBlocks(@TempDir final Path tempDir) throws Exception {
-    setupStorageWithHotBlocks(tempDir, 10);
+    final int blockCount = 10;
+    setupStorage(tempDir, blockCount, 0);
     try (final StorageSystem storage =
         context.createFileBasedStorage(
             spec, tempDir, StateStorageMode.PRUNE, StoreConfig.createDefault(), true)) {
       final Database database = storage.database();
-      assertHotBlockCounts(database.getColumnCounts(), 11, 11, 0, 0);
+      assertHotBlockCounts(database.getColumnCounts(), blockCount, blockCount, 0, 0);
       database.migrate();
       // payloads do get separated out, but there's only 1 default payload, so only 1 entry
-      assertHotBlockCounts(database.getColumnCounts(), 11, 0, 11, 1);
+      assertHotBlockCounts(database.getColumnCounts(), blockCount, 0, blockCount, 1);
+    }
+  }
+
+  @Test
+  void shouldMigrateFinalizedBlocks(@TempDir final Path tempDir) throws Exception {
+    final int totalBlocks = 39;
+    final int finalizedBlocks = 3 * spec.getGenesisSpec().getSlotsPerEpoch() + 1;
+    // hot storage has all hot blocks plus current finalized block
+    final int hotBlocks = totalBlocks - finalizedBlocks + 1;
```

Comment:
i need to have a small look at these numbers, just seems odd to be adding here

Reply:
I believe the current finalized block is stored in both hot and finalized storage which may explain it.